awq/entry.py

```diff
@@ -68,20 +77,16 @@ def build_model_and_enc(model_path):
     else:
         enc = AutoTokenizer.from_pretrained(model_path, use_fast=False)
 
+    kwargs = {"torch_dtype": torch.float16, "low_cpu_mem_usage": True}
+
     if args.load_quant:  # directly load quantized weights
-        # no need to really load the fp16 weights... just to get the model structure
         print("Loading pre-computed quantized weights...")
-        with init_empty_weights():
-            model = AutoModelForCausalLM.from_pretrained(model_path, config=config,
-                                                         torch_dtype=torch.float16, trust_remote_code=True)
+        model = AutoModelForCausalLM.from_pretrained(
```

Comment:
@Sakits: I had to remove `init_empty_weights` context block because no matter how many times I call `model.tie_weights()` after loading the real quantized weights, the `model.device` is always `meta` and this causes problems in `dispatch_model` to GPU down below.

This means we are unnecessarily loading fp16 weights in memory and then replacing them with quantized weights. But I see no other way to avoid this since `model.tie_weights` doesn't work.